### ISSUE TITLE
Inference with trained model produces null results on training and test images

### ISSUE THREAD

**Issue by PatrickGonzalez** [MIC-DKFZ/nnUNet#911]
Dear Fabian,

First and foremorst, thank you for all the wonderful work you have done on the nnUNet!

I have the following issue however:  I have a segmentation task on 2D rgb images: segmenting one retinal layer in polarization sensitive OCT images. I processed the images according to the Roadmap dataset example so that they are in the right format in the ImagesTr/ImagesTs/labelsTr/labelsTs folder. The prepocessing is then also done as per Roadmap example.

The 2D network runs for 50 epochs and I get very nice results in the validation set.

If I now try to run inference with the trained model on the test images and the training images (which include the validation images) I get null results only.

Any help would be very much appreciated.

Patrick González

**Comment by FabianIsensee** [comment]
Hey, are you absolutely certain that you processed the images you used for inference the same way as you did when you converted the dataset into nnU-Net format?
If that's the case I don't know how I could help you without actually looking at the dataset. Would that be possible?
Best,
Fabian

**Comment by PatrickGonzalez** [comment]
Dear Fabian,

Thank you for your swift reply. I understood that nnUNET_predic does the preprocessing so you can run it on the raw data? I included my raw_data folder for this specific task. It contains the training/test images and labels as well as the dataset.json file.

[Task106_OptAxis_Sclera.zip](https://github.com/MIC-DKFZ/nnUNet/files/7863647/Task106_OptAxis_Sclera.zip)
t

**Comment by PatrickGonzalez** [comment]
Hi Fabian,

wow! I re-trained the network with the workaround, and the inference runs as it should. Thank you for your great help!

Just to give you an idea: the images and segmentations come from a matlab .mat file. I noticed that in every scanmany pixels were nan's. I savesd this scans first as .png files with imageio.imsave after casting them to uint8. These png's where then preprocessed as in the road segmentation dataset. In the roadsegmentation conversion the segmentations are cast to uint32, the images aren't casted to any specific type.  I should probably cast them to float32 before saving them as pngs I guess.

Thank you for your help.

Best,

Patrick

**Comment by FabianIsensee** [comment]
Happy to hear that is works as intended now. I still have the feeling that something is off with your images. When you open the original images in the program provided by the OCT device do they still look like in my screenshot above?
Then I search for sclera OCT in google images the results I get look very different from your images:

![image](https://user-images.githubusercontent.com/11302164/149780751-03285cb9-7bd7-4f5c-8e81-3a1e963751b2.png)

I think you could be leaving a lot of segmentation performance on the table as long as the images are not the way they should be :-)

I am not sure whether png supports float values. Can't you just export as nifti from matlab? That should be possible

**Comment by PatrickGonzalez** [comment]
HI Fabian,

Thanks for the feedback. The images the network is trained on carry polarization information of the light reflected back by the various retinal layers. That is why they differ from the images you attached: those images are intensity images only. I'll have a look at png's and float values.

Best,
Patrick

**Comment by FabianIsensee** [comment]
I would probably avoid png images when dealing with floating point (and potentially negative) values ;-)

**Comment by FabianIsensee** [comment]
https://www.mathworks.com/help/images/ref/niftiwrite.html
